ChiModules/KEigenvalueSolver/IterativeMethods/power_iteration.cc

```diff
@@ -18,168 +18,127 @@ typedef sweep_namespace::SchedulingAlgorithm SchedulingAlgorithm;
 
 using namespace LinearBoltzmann;
 
-#include <iomanip>
 
 //###################################################################
 /**Power iterative scheme for k-eigenvalue calculations.
  * Note that this routine currently only works when the problem
  * is defined by a single groupset.
 */
-void KEigenvalue::Solver::PowerIteration(LBSGroupset& groupset)
+void KEigenvalue::Solver::PowerIteration()
 {
   chi_log.Log(LOG_0)
-    << "\n\n";
-  chi_log.Log(LOG_0)
-    << "********** Solving k-eigenvalue problem with "
-    << "the Power Method.\n\n";
-
-  // ----- Setting up required sweep chunks
-  auto sweep_chunk = SetSweepChunk(groupset);
-
-  // ----- Set sweep scheduler
-  MainSweepScheduler SweepScheduler(SchedulingAlgorithm::DEPTH_OF_GRAPH,
-                                    groupset.angle_agg,
-                                    *sweep_chunk);
-
-  // ----- Tool the sweep chunk
-  sweep_chunk->SetDestinationPhi(phi_new_local);
+      << "\n\n********** Solving k-eigenvalue problem with "
+      << "the Power Method.\n\n";
 
-  // ----- Set starting guess to a unit magnitude flux
-  phi_prev_local.assign(phi_prev_local.size(),1.0);
-  ScopedCopySTLvectors(groupset, phi_prev_local, phi_old_local);
+  phi_old_local.assign(phi_old_local.size(), 1.0);
 
-  // ----- Start outer k iterations
-  double F_prev = 1.0;       //Production source prev
-  double k_eff_prev = k_eff;
-  int nit = 0;               //number of iterations
-  bool k_converged = false;
+  double F_prev = 1.0;
+  double k_eff_prev = 1.0;
+  double k_eff_change = 1.0;
 
+  //================================================== Start power iterations
+  int nit = 0;
+  bool converged = false;
   while (nit < options.max_iterations)
   {
-    chi_log.Log(LOG_0VERBOSE_2)
-      << "\n********** Starting source iterations";
-
-
-    // ----- Start inner source iterations
-    double pw_change_prev = 1.0;
-    bool si_converged = false;
-    for (int si_nit=0; si_nit<groupset.max_iterations; si_nit++)
+    //============================================= Loop over groupsets
+    MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
What was the rationale behind this barrier?  I don't see a need for this here (but I might be missing something not included in the PR).  We need to be very careful with barrier placements.

Reply:
Ooooohhh. Please don't remove that one. He is probably mimicking LBS execute function (which will become AGS iteration). 

When things go wrong with graphs/cycles/FLUDS then that barrier has always given me a point of departure. It makes sense to have that synchronization point before starting to loop over group sets. We can maybe remove it later but for now please keep it. It only bothers performance when simulations are really small.

Pretty much just asking to keep my training wheels until I am a little more comfortable.